src/main/webapp/resources/js/apis/projects/samples.ts

```diff
@@ -35,18 +44,37 @@ export interface FieldUpdate {
   value: string;
 }
 
-export interface SampleRequest {
+export interface UpdateSampleItem extends CreateSampleItem {
+  sampleId: number;
+}
+
+export interface CreateSampleItem {
   name: string;
   organism?: string;
   description?: string;
   metadata: FieldUpdate[];
 }
 
+export interface UpdateSamplesRequest {
+  projectId: string;
+  body: UpdateSampleItem[];
+}
+
+export interface CreateSamplesRequest {
+  projectId: string;
+  body: CreateSampleItem[];
+}
+
 export interface ValidateSampleNamesRequest {
   samples: ValidateSampleNameModel[];
   associatedProjectIds?: number[];
 }
 
+export type SamplesFunction = (params: {
```

Comment:
Can you name the type something other than\`SamplesFunction\`, personally I like `CreateUpdateSamples`since it is doing both, but you can decide.  SampesFunction says nothing about its use

Reply:
Please see 8a388a5d8fa2188d67707266f29e6809bfbedcc0.